### PR TITLE
chore: migrate njlynch's modules to assertions

### PR DESCRIPTION
**Pull request by kaizencc**
This PR is complete when these 16 modules are accounted for:

- [x] route53
- [x] route53-targets
- [x] route53-patterns
- [x] elasticloadbalancingv2
- [x] elasticloadbalancingv2-targets
- [x] elasticloadbalancingv2-actions
- [x] elasticloadbalancing
- [x] cloudfront
- [x] cloudfront-origins
- [x] secretsmanager
- [x] ec2
- [x] certificates-manager
- [x] sns-subscriptions
- [x] sqs
- [x] ssm
- [x] kms

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-elasticloadbalancingv2/test/alb/target-group.test.ts [hunk 1]
```diff
@@ -38,13 +38,14 @@ describe('tests', () => {
       },
     });
 
-    expect(stack).not.toHaveResourceLike('AWS::ElasticLoadBalancingV2::TargetGroup', {
+    const matches = Template.fromStack(stack).findResources('AWS::ElasticLoadBalancingV2::TargetGroup', {
```
Comment (kaizen3031593) [inline packages/@aws-cdk/aws-elasticloadbalancingv2/test/alb/target-group.test.ts:41]: The original intention here is to do:

```ts
Template.fromStack(stack).hasResourceProperties('AWS::ElasticLoadBalancingV2::TargetGroup', Match.not({
  ...
});
```

@rix0rrr seems to hate this but I think it's reasonable; `Match.not` simply inverts the result so it succeeds with any matching failure. Curious if this is something you looked at and actively avoided.

**Review by kaizen3031593 — COMMENTED** [review]
Can't approve since I started this PR, but this is my stamp of approval :).

Just the one comment/question about `Match.not`...

File: packages/@aws-cdk/aws-elasticloadbalancingv2/test/alb/target-group.test.ts (hunk 1 above)
Comment (njlynch) [reply packages/@aws-cdk/aws-elasticloadbalancingv2/test/alb/target-group.test.ts:41]: I looked at it, it just felt... wrong? And it's not entirely equivalent; for example, the above fails if there are no TargetGroups. Maybe I should be using it, as I guess it's the more idiomatic approach for `assertions`, but something about the way the API currently expresses itself makes me wrinkle my nose a bit. I'd like to think of a better approach, but also didn't want to block this migration on it.

**Review by rix0rrr — APPROVED** [review]

File: packages/@aws-cdk/aws-ec2/test/vpc.test.ts [hunk 2]
```diff
@@ -1573,24 +1553,15 @@ describe('vpc', () => {
...
-            entry: {
-              type: 'aws:cdk:warning',
-              data: "fromVpcAttributes: 'availabilityZones' is a list token: the imported VPC will not work with constructs that require a list of subnets at synthesis time. Use 'Vpc.fromLookup()' or 'Fn.importListValue' instead.",
-            },
-          },
-        ),
-      ]));
-
-
+      expect(vpc.node.metadataEntry).toContainEqual({
```
Comment (corymhall) [inline packages/@aws-cdk/aws-ec2/test/vpc.test.ts:1560]: @kaizen3031593 this looks like the solution to what we were talking about yesterday.

File: packages/@aws-cdk/aws-elasticloadbalancingv2/test/alb/target-group.test.ts (hunk 1 above)
Comment (kaizen3031593) [reply packages/@aws-cdk/aws-elasticloadbalancingv2/test/alb/target-group.test.ts:41]: Thanks for the response. The equivalence issue is true, but to play devils advocate, I don't think the Assertions API was developed with the idea of exact 1-1 matching with assert-internal. But I hear what both you and Rico are saying.

File: packages/@aws-cdk/aws-ec2/test/vpc.test.ts (hunk 2 above)
Comment (kaizen3031593) [reply packages/@aws-cdk/aws-ec2/test/vpc.test.ts:1560]: Good call. Although I am about to write a PR for messages support that would look something like this:

```ts
AssertAnnotations.fromStack(stack).hasWarning('*', {
  entry: {
    type: 'aws:cdk:warning',
    data: "fromVpcAttributes...",
  },
});
```

**Pull request merged**